audio/stereo_panner.rs

```diff
@@ -0,0 +1,132 @@
+use block::{Block, Chunk, Tick, FRAMES_PER_BLOCK};
+use euclid::Vector3D;
+use node::{AudioNodeEngine, AudioNodeMessage, BlockInfo};
+use node::{AudioNodeType, ChannelInfo};
+use param::{Param, ParamDir, ParamType};
+use std::f32::consts::PI;
+
+// .normalize(), but it takes into account zero vectors
+use panner_node::normalize_zero;
+
+#[derive(Copy, Clone, Debug)]
+pub struct StereoPannerOptions {
+    pub pan: f32,
+}
+
+impl Default for StereoPannerOptions {
+    fn default() -> Self {
+        StereoPannerOptions {
+            pan: 0.,
+        }
+    }
+}
+
+#[derive(AudioNodeCommon)]
+pub(crate) struct StereoPannerNode {
+    channel_info: ChannelInfo,
+    pan: Param,
+}
+
+impl StereoPannerNode {
+    pub fn new(options: StereoPannerOptions, channel_info: ChannelInfo) -> Self {
+        Self {
+            Param::new(options.pan),
+        }
+    }
+
+    pub fn update_parameters(&mut self, info: &BlockInfo, tick: Tick) -> bool {
+        self.pan.update(info, tick)
+    }
+}
+
+impl AudioNodeEngine for StereoPannerNode {
+    fn node_type(&self) -> AudioNodeType {
+        AudioNodeType::StereoPannerNode
+    }
+
+    fn process(&mut self, mut inputs: Chunk, info: &BlockInfo) -> Chunk {
+        debug_assert!(inputs.len() == 1);
+
+        return inputs;
+
+        {
+            let block = &mut inputs.blocks[0];
+
+            block.explicit_repeat();
+
+            let mono = if block.chan_count() == 1 {
+                block.resize_silence(2);
+                true
+            } else {
+                debug_assert!(block.chan_count() == 2);
+                false
+            };
+
+                let (l, r) = block.data_mut().split_at_mut(FRAMES_PER_BLOCK.0 as usize);
+                for frame in 0..FRAMES_PER_BLOCK.0 {
+                    let frame = Tick(frame);
+                    if self.update_parameters(info, frame) {
+                    pan = self.pan.value();
```

Comment:
You need to initialize this outside of the loop with `let pan = self.pan.value();`

Also, please run `rustfmt` on this file.